go/client/cmd_simplefs_list.go

```diff
@@ -95,28 +198,58 @@ func (c *CmdSimpleFSList) Run() error {
 		if err != nil {
 			return err
 		}
+		gotList := false
 		for {
 			listResult, err := cli.SimpleFSReadList(ctx, opid)
+			// Eat the error here because it may just mean the results
+			// are complete. TODO: should KBFS return non-error here
+			// until the opid is closed?
 			if err != nil {
-				break
+				if gotList == true {
+					err = nil
+				}
+				return err
+			}
+			gotList = true
+			err = c.output(listResult)
+			if err != nil {
+				return err
 			}
-			c.output(listResult)
 		}
 	}
 	return err
 }
 
-func (c *CmdSimpleFSList) output(listResult keybase1.SimpleFSListResult) {
-
+func (c *CmdSimpleFSList) output(listResult keybase1.SimpleFSListResult) error {
 	ui := c.G().UI.GetTerminalUI()
 
-	for _, e := range listResult.Entries {
-		if e.DirentType == keybase1.DirentType_DIR {
-			ui.Printf("%s\t<%s>\t\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Name)
-		} else {
-			ui.Printf("%s\t%s\t%d\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Size, e.Name)
+	if c.winStyle {
+		for _, e := range listResult.Entries {
+			if e.DirentType == keybase1.DirentType_DIR {
+				ui.Printf("%s\t<%s>\t\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Name)
+			} else {
+				ui.Printf("%s\t%s\t%d\t%s\n", keybase1.FormatTime(e.Time), keybase1.DirentTypeRevMap[e.DirentType], e.Size, e.Name)
+			}
+		}
+	} else {
+		// capture the current terminal dimensions
+		terminalWidth, _, err := terminal.GetSize(int(os.Stdout.Fd()))
+		if err != nil {
+			return err
+		}
+
+		var outputBuffer bytes.Buffer
+
+		err = c.ls(&outputBuffer, listResult, terminalWidth)
+		if err != nil {
+			return err
+		}
+
+		if outputBuffer.String() != "" {
+			ui.Printf("%s", outputBuffer.String())
```

Comment:
but i don't know what would be adding the extra newline and none of the other commands have this!